crates/starknet-devnet-server/src/api/json_rpc/spec_reader/mod.rs

```diff
@@ -318,13 +311,29 @@ mod tests {
                                 StarknetResponse::AddInvokeTransaction(_)
                             ));
                         }
-                        _ => {
-                            // Remaining responses are not implemented, because
-                            // multiple requests return the same response format either u64, Felt,
-                            // etc. so its impossible to know which
-                            // response variant is generated based on
-                            // serde untagged deserialization. This is due to the fact that the
-                            // first variant which complies with the response format is returned
+                        StarknetRequest::SpecVersion | StarknetRequest::ChainId => {
+                            assert!(matches!(
+                                sn_response,
+                                StarknetResponse::String(_) | StarknetResponse::Felt(_)
+                            ));
+                        }
```

Comment:
Its because chain id is returned as 0x prefixed hex string, which in fact is a string. Maybe better solution is for starknet_chainId RPC method the return type to be felt instead of string.

Reply:
I meant why isn't it matched like this:
```rust
                        StarknetRequest::SpecVersion => {
                            assert!(matches!(sn_response, StarknetResponse::String(_)));
                        }
                        StarknetRequest::ChainId => {
                            assert!(matches!(sn_response, StarknetResponse::Felt(_)))
                        }
```

Reply:
[9659369](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/411/commits/9659369a484d94a3e27c77f07c59084c203fd851)